searchv2/rules.py

```diff
@@ -0,0 +1,405 @@
+from datetime import timedelta
+from django.utils import timezone
+from pydantic import BaseModel
+from typing import List
+from typing import Optional
+
+from package.models import Package
+
+
+class CheckResult(BaseModel):
+    """
+    Model to store the result of a check, which includes the score and a message.
+    """
+
+    score: int
+    message: str
+
+
+class ScoreRule(BaseModel):
+    """
+    Abstract base class for all score rules. It validates the input data using Pydantic,
+    and requires a `check` method that should be implemented by subclasses.
+
+    Attributes:
+        name: The name of the score rule.
+        description: The description of the score rule.
+        max_score: The maximum score that can be awarded by this rule.
+        documentation_url: An optional URL to the documentation for this rule.
```

Comment:
I'll rebuild the requirements too in case that was no helping too.

Reply:
> For the remaining todo items, would it make sense to break those into their own issues, or keep them all combined here?

I'm open to ideas. It could make sense to build out new issues. It might be worth mentioning that this code isn't actually getting called by search yet. 

> Also, there was a weird thing with `pydantic` being added. I kept getting an error on `pydantic` not being available in the container `django-q`. I eventually solved it by removing the image that was having the issue and running `just up` again.
> 
> I didn't see anywhere in the docs on how to 'refresh' the containers. I'm not sure the best way to approach that.

When the requirements.* changes we have to rebuild the container. I could make the just command do a rebuild on `just up` but that can be annoying if you don't want it to. Not sure if a few `echo "Please remember to rebuild if you see ... message"` would be helpful or just in the way. Thoughts?

Reply:
I think a reminder with the `just up` should be sufficient in those cases, though if someone runs it in detached they wouldn’t see it 🤔

Maybe the [FAQ](https://docs.djangopackages.org/en/latest/faq/) can be updated with a question like

> Question: I’m getting a missing dependency when trying to run locally. What do i do? 
> Answer: Rebuild the django docker image by running <insert rebuild command here>